modin/core/execution/ray/generic/io/io.py

```diff
@@ -18,3 +18,15 @@
 
 class RayIO(BaseIO):
     """Base class for doing I/O operations over Ray."""
+
+    @classmethod
+    def from_ray(cls, ray_obj):
+        raise NotImplementedError(
+            f"Modin dataset cann't be created from `ray.data.Dataset` using {cls}."
+        )
```

Comment:
Why? Is the PR in draft?

Reply:
The default implementation (in the BaseIO class) throws an exception stating that _the Ray Engine is required for the `to_ray`/`from_ray` functions_. We also have an additional base class io for the Ray engine that will be used for every data structure implementation using the Ray engine, so we expect all its subclasses to implement these functions in their own way. If we create a new subclass of RayIO and forget to implement these functions, we will get a Not Implemented exception.

In this PR I hasn't added the ``to_ray`/`from_ray` implementation to ``cudf_on_ray`` because it works incorrectly if I'm not mistaken. If we try to use it and call `to_ray`/`from_ray` functions, we will also receive a Not Implemented exception.